### ISSUE TITLE
git curl fails when remote terminates unexpectedly

### ISSUE THREAD

**Issue by DavidIMcIntosh** [git-for-windows/git#2012]
- [ x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?

```
$ git --version --build-options

git version 2.20.1.windows.1

```
64-bit version.  sorry, uninstalled it before I knew to add --build-options switch
This does NOT happen with v 2.19.1 or earlier.

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

```
$ cmd.exe /c ver

Microsoft Windows [Version 6.1.7601]
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

```
# One of the following:
> type "C:\Program Files\Git\etc\install-options.txt"

Editor Option: Notepad++
Custom Editor Path: 
Path Option: Cmd
Plink Path: C:\Program Files\TortoiseGit\bin\TortoiseGitPlink.exe
SSH Option: Plink
CURL Option: OpenSSL
CRLF Option: CRLFAlways
Bash Terminal Option: MinTTY
Performance Tweaks FSCache: Enabled
Use Credential Manager: Enabled
Enable Symlinks: Enabled
Enable Builtin Rebase: Disabled
Enable Builtin Stash: Disabled
```

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

Corporate environment with altogether toooo many security processes complicating issues, such as hidden proxy setup, etc.

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

Many, but for fundamental testing, cmd.exe prompt.  Issue is the same whether run from cmd.exe, SourceTree wrapper, TortoiseGIT wrapper, etc.

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
cd /d <git_dir>
set GIT_CURL_VERBOSE=1
git  pull>C:\temp\gitOutput.txt
```
 - What did you expect to occur after running these commands?

git pull

 - What actually happened instead?
Pull (or fetch or ...) consistently fails with "remote end hung up unexpectedly".
We are using Bonobo hosted on IIS on the server side, and using Windows single-sign-on authentication from corporate AD server.
See attached gitOutput_redacted.txt.
[gitOutput_redacted.txt](https://github.com/git-for-windows/git/files/2738745/gitOutput_redacted.txt)

 - If the problem was occurring with a specific repository, can you provide the
   URL to that repository to help us with testing?

Server is internal and not visible beyond our firewall.  We are using Bonobo hosted on IIS on the server side, and using Windows single-sign-on authentication from corporate AD server.

**Comment by dscho** [comment]
>     * If the problem was occurring with a specific repository, can you provide the
>       URL to that repository to help us with testing?

Since you cannot offer a reliable way to reproduce your problem, I expect you will be stuck with having to investigate this from start to finish, with hopefully helpful suggestions thrown in by others.

I could imagine that using empty credentials in the URL could help. And `GIT_TRACE_CURL` might offer more useful information.

**Comment by DavidIMcIntosh** [comment]
Re: "provide the URL to that repository" - its an internal repository, so not possible (note the plethora of "Redacted"'s in the attached log file).
We are OK here, we just reverted to 2.19.1.  This was filed mostly as an FYI for developers, in case it helps anyone - its clearly an issue, but may be minor; time will tell.
If I can be of help, drop me a note and I will be glad to comply.
I think I tried empty credentials, but if I have time, I will re-install and double-check.  Will use GIT_TRACE_CURL at that time too.  Thanks.
